src/main/java/sample/stream/TcpEcho.java

```diff
@@ -61,14 +57,13 @@ public static void main(String[] args) {
   public static void server(ActorSystem system, InetSocketAddress serverAddress) {
     final ActorFlowMaterializer materializer = ActorFlowMaterializer.create(system);
 
-    final Sink<IncomingConnection, Future<BoxedUnit>> handler = Sink.foreach(conn -> {
+    final Sink<IncomingConnection, Future<BoxedUnit>> handler = Sink.runForeach(conn -> {
       System.out.println("Client connected from: " + conn.remoteAddress());
       conn.handleWith(Flow.<ByteString>empty(), materializer);
     });
 
-    final Future<ServerBinding> bindingFuture = 
-        StreamTcp.get(system)
-          .bind(serverAddress).to(handler).run(materializer);
+    final Future<ServerBinding> bindingFuture =
+      Tcp.get(system).bind(serverAddress.getHostString(), serverAddress.getPort()).to(handler).run(materializer);
```

Comment:
getHostString this requires java7, I guess that is fine, but we could also skip creating the InetSocketAddress in the first place